ksql-engine/src/main/java/io/confluent/ksql/KsqlContext.java

```diff
@@ -46,24 +46,43 @@
   private final AdminClient adminClient;
   private final KafkaTopicClient topicClient;
 
-  public static KsqlContext create(Map<String, Object> streamsProperties)
+  public static KsqlContext create(KsqlConfig ksqlConfig)
       throws ExecutionException, InterruptedException {
-    if (streamsProperties == null) {
-      streamsProperties = new HashMap<>();
+    if (ksqlConfig == null) {
+      ksqlConfig = new KsqlConfig(Collections.emptyMap());
     }
+    Map<String, Object> streamsProperties = ksqlConfig.getKsqlStreamConfigProps();
     if (!streamsProperties.containsKey(StreamsConfig.APPLICATION_ID_CONFIG)) {
       streamsProperties.put(StreamsConfig.APPLICATION_ID_CONFIG, APPLICATION_ID_OPTION_DEFAULT);
     }
     if (!streamsProperties.containsKey(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG)) {
       streamsProperties.put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, KAFKA_BOOTSTRAP_SERVER_OPTION_DEFAULT);
     }
-    KsqlConfig ksqlConfig = new KsqlConfig(streamsProperties);
     AdminClient adminClient = AdminClient.create(ksqlConfig.getKsqlAdminClientConfigProps());
     KafkaTopicClient topicClient = new KafkaTopicClientImpl(adminClient);
     KsqlEngine ksqlEngine = new KsqlEngine(ksqlConfig, topicClient);
     return new KsqlContext(adminClient, topicClient, ksqlEngine);
   }
 
+  // For test purpose
+  public static KsqlContext create(KsqlConfig ksqlConfig, SchemaRegistryClient schemaRegistryClient)
+      throws ExecutionException, InterruptedException {
+    if (ksqlConfig == null) {
```

Comment:
This code is nearly identical to the other create method, except that we don't pass the `schemaRegistryClient` when creating the KSQL engine. Would be better to deduplicate.

Reply:
Yes, removed duplicate code.